csrc/quantization/fused_kernels/quant_conversions.cuh

```diff
@@ -31,9 +31,12 @@ static __device__ __forceinline__ int8_t float_to_int8_rn(float const x) {
 #endif
 }
 
-static __device__ __forceinline__ FP8_TYPE float_to_fp8(float const x) {
-  float const r = fmax(-FP8_E4M3_MAX, fmin(x, FP8_E4M3_MAX));
-  return static_cast<FP8_TYPE>(r);
+template <typename fp8_type>
+static __device__ __forceinline__ void float_to_fp8(fp8_type& ret,
```

Comment:
Same here, curious why you changed this to update by reference?

Reply:
Reverted.